sql/core/src/test/scala/org/apache/spark/sql/DynamicPartitionPruningSuite.scala

```diff
@@ -423,7 +423,7 @@ class DynamicPartitionPruningSuite
           """
             |SELECT * FROM fact_sk f
             |JOIN dim_store s
-            |ON f.date_id = s.store_id
+            |ON f.store_id = s.store_id
```

Comment:
Why does this change match the description "no predicate on the dimension table"?

Reply:
If we don't change it. `partScan` will always be None. `hasPartitionPruningFilter` will be false If we change it. It will match "no predicate on the dimension table"
```
var partScan = getPartitionTableScan(l, left)
if (partScan.isDefined && canPruneLeft(joinType) &&
    hasPartitionPruningFilter(right)) {
    val hasBenefit = pruningHasBenefit(l, partScan.get, r, right)
    newLeft = insertPredicate(l, newLeft, r, right, rightKeys, hasBenefit)
} else {
    partScan = getPartitionTableScan(r, right)
    if (partScan.isDefined && canPruneRight(joinType) &&
        hasPartitionPruningFilter(left) ) {
        val hasBenefit = pruningHasBenefit(r, partScan.get, l, left)
        newRight = insertPredicate(r, newRight, l, left, leftKeys, hasBenefit)
    }
}
```

Reply:
Yes. This should be a join on the partitioned column, `fact_sk.store_id`, because the previous test covers `not a partition column` case. And, `no predicate on the dimension table` should mean `WHERE s.country = 'NL'`.